lib/Parser/Parse.cpp

```diff
@@ -1398,15 +1398,52 @@ ParseNodePtr Parser::AddVarDeclNode(IdentPtr pid, ParseNodePtr pnodeFnc)
     return pnode;
 }
 
-ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr pid)
+Js::PropertyId Parser::EnsurePropertyId(IdentPtr pid)
 {
-    ParseNodePtr declNode = CreateBlockScopedDeclNode(pid, knopConstDecl);
+    Js::PropertyId propertyId = pid->GetPropertyId();
+    if (propertyId == Js::Constants::NoProperty)
+    {
+        propertyId = m_scriptContext->GetOrAddPropertyIdTracked(pid->Psz(), pid->Cch());
+        pid->SetPropertyId(propertyId);
+    }
+    return propertyId;
+}
 
-    declNode->sxVar.sym->SetIsModuleExportStorage(true);
+ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr localName)
+{
+    ParseNodePtr declNode = CreateBlockScopedDeclNode(localName, knopConstDecl);
+    Symbol* sym = declNode->sxVar.sym;
+    
+    sym->SetIsModuleExportStorage(true);
 
     return declNode;
 }
 
+Symbol* Parser::MarkIdentifierReferenceIsModuleExport(IdentPtr localName)
+{
+    PidRefStack* pidRef = this->PushPidRef(localName);
```

Comment:
That's sort of what I'm trying to do. This method is trying to find the symbol for localName so that we can mark it as stored in an export slot. We use this method when we're exporting something via a binding identifier instead of having the object itself. 

An example:
function foo() { };
export { foo };
export { foo as bar };

In both export statements here, we use MarkIdentifierReferenceIsModuleExport to find and mark the symbol for identifier 'foo' as being an export.